cypress/e2e/dashboard_atomic_tests.cy.js

```diff
@@ -0,0 +1,185 @@
+      .clickRegister();
+    new LoginPage()
+      .openTegb()
+      .typeUsername(username)
+      .typePassword(password)
+      .clickLogin();
+  });
+
+  context("Header tests", () => {
+    it("Logo is visible", () => {
+      new HomePage().logo.isVisible();
+    });
+
+    it("Title is visible", () => {
+      new HomePage().pageTitle.isVisible();
+    });
+
+    it("Title have text", () => {
+      new HomePage().pageTitle.hasText("TEG#B Dashboard");
+    });
+
+    it("Logout button is visible", () => {
+      new HomePage().logoutButton.isVisible();
+    });
+
+    it("Logout button has text", () => {
+      new HomePage().logoutButton.hasText("Odhlásit se");
+    });
+  });
+
+  context("Menu tests", () => {
+    it("Home tab is visible", () => {
+      new HomePage().homeMenu.isVisibleXpath();
+    });
+
+    it("Home tab has text", () => {
+      new HomePage().homeMenu.hasTextXpath("Domů");
+    });
+
+    it("Accounts tab is visible", () => {
+      new HomePage().accountsMenu.isVisibleXpath();
+    });
+
+    it("Accounts tab has text", () => {
+      new HomePage().accountsMenu.hasTextXpath("Účty");
+    });
+
+    it("Transactions tab is visible", () => {
+      new HomePage().transactionsMenu.isVisibleXpath();
+    });
+
+    it("Transactions tab has text", () => {
+      new HomePage().transactionsMenu.hasTextXpath("Transakce");
+    });
+
+    it("Support tab is visible", () => {
+      new HomePage().supportMenu.isVisibleXpath();
+    });
+
+    it("Support tab has text", () => {
+      new HomePage().supportMenu.hasTextXpath("Podpora");
+    });
+  });
+
+  context("Footer tests", () => {
+    it("Footer is visible", () => {
+      new HomePage().footer.isVisible();
+    });
+
+    it("Footer have text", () => {
+      new HomePage().footer.hasText("© 2023 Banking App");
+    });
+  });
+
+  context("Profile details on Dashboard tests", () => {
+    it("Profile Details header is visible", () => {
+      new HomePage().profileDetailHeader.isVisible();
+    });
+
+    it("Profile Details header has text", () => {
+      new HomePage().profileDetailHeader.hasText("Detaily Profilu");
+    });
+
+    it("Edit Profile Buttion is visible", () => {
+      new HomePage().editProfileButton.isVisible();
+    });
+
+    it("Edit Profile Button has text", () => {
+      new HomePage().editProfileButton.hasText("Upravit profil");
+    });
+
+    it("First Name has attribute", () => {
+      new HomePage().firstName.hasAttribute("data-testid", "name");
+    });
+
+    it("Last Name has attribute", () => {
+      new HomePage().lastName.hasAttribute("data-testid", "surname");
+    });
+
+    it("Email has attribute", () => {
+      new HomePage().email.hasAttribute("data-testid", "email");
+    });
+
+    it("Phone Number has attribute", () => {
+      new HomePage().phoneNumber.hasAttribute("data-testid", "phone");
+    });
+
+    it("Age has attribute", () => {
+      new HomePage().age.hasAttribute("data-testid", "age");
```

Comment:
Atributy tady úplně asi kontrolovat nemusíš, data-testid jsou pro psaní automatů, pro business až tak důležité nejsou